testutils/clustermanager/gke.go

```diff
@@ -235,24 +246,36 @@ func (gc *GKECluster) Acquire() error {
 		}
 
 		clusterLoc := getClusterLocation(region, gc.Request.Zone)
-		// TODO(chaodaiG): add deleting logic once cluster deletion logic is done
 
-		log.Printf("Creating cluster %q' in %q", clusterName, clusterLoc)
-		var createOp *container.Operation
-		createOp, err = gc.operations.create(*gc.Project, clusterLoc, rb)
+		// Deleting cluster if it already exists
+		existingCluster, _ := gc.operations.get(*gc.Project, clusterLoc, clusterName)
+		if nil != existingCluster {
+			log.Printf("Cluster %q already exists in %q. Deleting...", clusterName, clusterLoc)
+			op, err = gc.operations.delete(*gc.Project, clusterName, clusterLoc)
+			if nil == err {
+				err = gc.wait(clusterLoc, op.Name, deletionTimeout)
+			}
+		}
+		// Creating cluster only if previous step succeeded
 		if nil == err {
-			if err = gc.wait(clusterLoc, createOp.Name, creationTimeout); nil == err {
-				cluster, err = gc.operations.get(*gc.Project, clusterLoc, rb.Cluster.Name)
+			log.Printf("Creating cluster %q in %q", clusterName, clusterLoc)
+			op, err = gc.operations.create(*gc.Project, clusterLoc, rb)
+			if nil == err {
+				if err = gc.wait(clusterLoc, op.Name, creationTimeout); nil == err {
+					cluster, err = gc.operations.get(*gc.Project, clusterLoc, rb.Cluster.Name)
+				}
 			}
 		}
 		if nil != err {
-			errMsg := fmt.Sprintf("error creating cluster: '%v'", err)
+			errMsg := fmt.Sprintf("Error creating cluster: '%v'. ", err)
```

Comment:
It could be a delete error, so better making the message generic.

```suggestion
			errMsg := fmt.Sprintf("Error during cluster creation: '%v'. ", err)
```